### ISSUE TITLE
Use the function keys for MLOAD 0-9 ?

### ISSUE THREAD

**Issue by dragon788** [Subsystems-us/SMART-Response-XE-Tiny-Basic-Port#1]
This is a less ergonomic device for writing than the Alphasmart Neo/Neo2, but if the function keys could save the current file in whatever slot it was loaded from, and then switch to a new slot, that would go a long way towards allowing for modular programs or just writing some prose notes separate from programs.

Perhaps it could store the SAVESLOT as a variable at the top of the file to make it easier to know "where" a file exists?

**Comment by GeneralStingray** [comment]
I am having difficulty getting the Smart Response to give valid feedback to the programmer.  I am working with an IVRASP Mk2 on a Debian machine.  I have been successful programming Arduino boards with this setup but with the Smart Response I am getting the following:

avrdude: Device signature = 0xe047d0
avrdude: Expected signature for ATmega128RFA1 is 1E A7 01

I am programming it with the following (note that I copied the .hex to TBSE.hex to reduce the number of keystrokes):

avrdude -P /dev/ttyUSB0 -c avrisp2 -p ATmega128RFA1 -U flash:w:TBSE.hex -F

I have batteries installed in the Smart Response XE while doing this because I gather the AVRISP Mk2 doesn't supply power.

I would appreciate any suggestions you can offer.

**Comment by Subsystems-us** [comment]
You are right, the AVRISP does not supply power, but it looks for it to be
present before it will let you program. The fact that the programmer is
reading the wrong part number points to a problem with the communication
link or, less likely, a corrupt signature. Try using the -F switch to
override the signature check and see if it will program.

I have a MkII and use Microchip Studio and usually don't have an issue.

Dan

**Comment by Subsystems-us** [comment]
Ooops, sorry I missed that you are using the -F. Anytime I have had issues
with programming, a lot of time it has been the USB driver. Check to see if
you have the Juno or libusb driver loaded (should be libusb). Also, I think
you could try just the -P USB to let avrdude look for and configure the USB
port. I am not a big Linux guy so I am not too much help on the OS
specifics.
